### PR TITLE
[BCF-2689] Use an in-memory keystore for tests

### PR DESCRIPTION
**Pull request by cedric-cordenier**
- Remove unused migration methods from the keystore
- Split the keystore ORM in two, between methods which access the
  `encrypted_key_rings` table, and methods which access `evm.key_states`
- Create an in-memory version of the base ORM, which doesn't actually
  write to the DB. Methods which access `evm.key_states` are unaffected
  by this. This will prevent tests from competing to get a row-level
  lock on the singleton keystore record.
- Start using this in-memory keystore in `cltest` and in tests
  generally. Note: tests in the keystore package themselves are
  unaffected by this, since we still want to have test coverage over the
  behaviour that accesses the DB.

### CODE REVIEW

File: core/services/keystore/master.go [hunk 1]
```diff
@@ -141,17 +144,21 @@ func (ks *master) VRF() VRF {
...
+type ORM interface {
+	isEmpty() (bool, error)
+	saveEncryptedKeyRing(*encryptedKeyRing, ...func(pg.Queryer) error) error
+	getEncryptedKeyRing() (encryptedKeyRing, error)
+}
+
+type keystateORM interface {
+	loadKeyStates() (*keyStates, error)
+	getNextNonce(common.Address, *big.Int, ...pg.QOpt) (int64, error)
+	incrementNextNonce(common.Address, *big.Int, int64, ...pg.QOpt) (int64, error)
```
Comment (jmank88) [inline core/services/keystore/master.go:156]: Are these still used?

File: core/services/keystore/keystoretest.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+func (o *memoryORM) getEncryptedKeyRing() (encryptedKeyRing, error) {
+	o.mu.RLock()
+	defer o.mu.RUnlock()
+	if o.keyRing == nil {
+		return encryptedKeyRing{}, nil
+	}
+	return *o.keyRing, nil
+}
+
+func NewInMemoryORM(q pg.Queryer) *memoryORM {
```
Comment (jmank88) [inline core/services/keystore/keystoretest.go:48]: Does this need to be exported?

File: core/services/keystore/master.go (hunk 1 above)
Comment (jmank88) [reply core/services/keystore/master.go:156]: Oh I was thinking that #10108 had been merged. Does this become simpler after that PR?
- https://github.com/smartcontractkit/chainlink/pull/10108

Comment (cedric-cordenier) [reply core/services/keystore/master.go:156]: Yes, they're used from within `keystore.Eth()`.

File: core/services/keystore/keystoretest.go (hunk 2 above)
Comment (cedric-cordenier) [reply core/services/keystore/keystoretest.go:48]: Nope, good catch 👍

File: core/services/keystore/master.go (hunk 1 above)
Comment (cedric-cordenier) [reply core/services/keystore/master.go:156]: @jmank88 Ah, I hadn't seen that PR: it looks like keystateORM would only include the loadKeyStates method, so it is a simplification, yes.

**Review by george-dorin — DISMISSED** [review]

**Pull request closed without merging**